src/main/java/gov/nasa/pds/validate/commandline/options/ConfigKey.java

```diff
@@ -159,4 +164,26 @@ public class ConfigKey {
   public static final String TARGET_MANIFEST = "validate.targetManifest";
 
   public static final String SKIP_PRODUCT_VALIDATION = "validate.ignoreProductValidation";
+  
+  /* dumb way to fix this, but this class should have been enum to start with and now it is too late
+   * 
+   * Declare a Set then put all the values in it. This way we can check that values in the
+   * config file are allowable. As currently written it is simple to expand this list but
+   * not include an update later checks if it is valid or not. Really, this is just a smelly
+   * code hack for bad design choice back when it was a quick and dirty tool.
+   */
+  public static final Set<String> ALL_KEYWORDS;
+  static {
+    ALL_KEYWORDS = new HashSet<String>();
+    for (Field field : ConfigKey.class.getDeclaredFields()) {
+      if (Modifier.isStatic(field.getModifiers()) && field.getType().equals(String.class)) {
+        try {
+          ALL_KEYWORDS.add(field.get(null).toString());
+        } catch (IllegalArgumentException | IllegalAccessException e) {
+          // can we every get here??
+          e.printStackTrace();

```

Comment:
`every` → `even`

However, if getting "here" is an unexpected situation, maybe aborting is the right approach rather than just printing the stack trace and soldering on? Maybe throw a new `IllegalStateException`. Since that's a `RuntimeException` you don't have to declare it.

Thinking further: `IllegalArgumentException` is itself a `RuntimeException` so you don't even have to catch it. Only `IllegalAccessException` needs to be handled—and maybe aborting if it occurs is the right thing? I dunno 🤷

Reply:
Excellent suggestion. Done.